lua/entities/gmod_wire_expression2/base/compiler.lua

```diff
@@ -651,42 +651,57 @@ end
 
 function Compiler:InstrNUM(args)
 	self.prfcounter = self.prfcounter + 0.5
-	RunString("E2Lib.Compiler.native = function() return " .. args[3] .. " end")
-	return { Compiler.native }, "n"
+	local num = tonumber(args[3])
```

Comment:
This looks like an okay change, because `tonumber` parses numbers the same as Lua's own parser, and we only call this function with things that look numeric.